substrafl_examples/get_started/plot_substrafl_sklearn_fedavg.py

```diff
@@ -0,0 +1,571 @@
+        data_sample,
+        local=True,
+    )
+
+    # Add the testing data on each organization.
+    data_sample = DataSampleSpec(
+        data_manager_keys=[dataset_keys[org_id]],
+        test_only=True,
+        path=data_path / f"org_{ind+1}" / "test",
+    )
+    test_datasample_keys[org_id] = client.add_data_sample(
+        data_sample,
+        local=True,
+    )
+
+# %%
+# Metrics registration
+# ====================
+#
+# A metric corresponds to an algorithm used to compute the score of predictions on a
+# **datasample**.
+# Concretely, a metric corresponds to an archive *(tar or zip file)*, automatically build
+# from:
+#
+# - a **Python scripts** that implement the metric computation
+# - a `Dockerfile <https://docs.docker.com/engine/reference/builder/>`__ to specify the required dependencies of the
+#   **Python scripts**
+
+import zipfile
+
+from substra.sdk.schemas import AlgoInputSpec
+from substra.sdk.schemas import AlgoOutputSpec
+from substra.sdk.schemas import AlgoSpec
+from substra.sdk.schemas import AssetKind
+
+permissions_metric = Permissions(public=False, authorized_ids=[ALGO_ORG_ID] + DATA_PROVIDER_ORGS_ID)
+
+inputs_metrics = [
+    AlgoInputSpec(
+        identifier="datasamples",
+        kind=AssetKind.data_sample,
+        optional=False,
+        multiple=True,
+    ),
+    AlgoInputSpec(identifier="opener", kind=AssetKind.data_manager, optional=False, multiple=False),
+    AlgoInputSpec(identifier="predictions", kind=AssetKind.model, optional=False, multiple=False),
+]
+
+outputs_metrics = [AlgoOutputSpec(identifier="performance", kind=AssetKind.performance, multiple=False)]
+
+objective = AlgoSpec(
+    inputs=inputs_metrics,
+    outputs=outputs_metrics,
+    name="Accuracy",
+    description=assets_directory / "metric" / "description.md",
+    file=assets_directory / "metric" / "metrics.zip",
+    permissions=permissions_metric,
+)
+
+METRICS_DOCKERFILE_FILES = [
+    assets_directory / "metric" / "iris_metrics.py",
+    assets_directory / "metric" / "Dockerfile",
+]
+
+archive_path = objective.file
+with zipfile.ZipFile(archive_path, "w") as z:
+    for filepath in METRICS_DOCKERFILE_FILES:
+        z.write(filepath, arcname=filepath.name)
+
+metric_key = clients[ALGO_ORG_ID].add_algo(objective)
+
+# %%
+# Specify the machine learning components
+# ***************************************
+#
+# In this section, you will register an algorithm and its dependencies,
+# and specify the federated learning strategy
+# as well as the nodes on which to train and to test.
+
+# %%
+# Model definition
+# ================
+#
+# The machine learning model used here is a logistic regression.
+# The `warm_start` argument is essential is this example as it indicate to use the current state of the model
+# as initialization for the future training.
+# Also, we choose to set `max_iter` to 3 to limit the `num_updates` between aggregation in order to maximize
+# the communication between organizations.
```

Comment:
- How does it limit the "num_updates"? Actually do we have a "num_update" concept without PyTorch?
- We should explain why we are interested here to maximise the communication between orgs

Reply:
I rephrased, tell me if it's better.